containers/src/Data/Sequence.hs

```diff
@@ -102,9 +102,15 @@
 --
 -- Several functions take special advantage of sharing to produce
 -- results using much less time and memory than one might expect. These
--- are documented individually for functions, but also include the
--- methods '<$' and '*>', each of which take time and space proportional
--- to the logarithm of the size of the result.
+-- are documented individually for functions, but also include certain
+-- class methods:
+--
+-- '<$' and '*>' each take time and space proportional
+-- to the logarithm of the size of their result.
+--
+-- '<*' takes time and space proportional to the product of the length
+-- of its first argument and the logarithm of the length of its second
+-- argument.
```

Comment:
Huh. Why do the asymptotics for `<*` and `*>` differ? Is that really right?

Reply:
Yes, but it needs more explanation. `*>` gets to reuse the original structure of the second sequence quite heavily, so that's correct for *additional* allocation and residency, but doesn't count the fact that it will retain large portions of the original structure of the second sequence as well. Please feel free to suggest a better blurb. For `<*`, we need to build an all new copy of a structure like the second sequence for every element of the first sequence.

Reply:
For `*>`, once we've turned the second sequence inside out (in time logarithmic in its length), we just stick that in a bunch of places unchanged. We can't get that kind of repetition in `<*`.